src/Compilers/Core/Portable/Syntax/InternalSyntax/ChildSyntaxList.cs

```diff
@@ -1,5 +1,7 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
 namespace Microsoft.CodeAnalysis.Syntax.InternalSyntax
 {
     internal partial struct ChildSyntaxList
```

Comment:
❔ Shouldn't the constructor allow null here?

Reply:
I agree the field should allow null but not the constructor. 

The field should be nullable because this is a `struct` and hence that is a valid state. For the constructor though, that is explicitly constructing the object, I'd expect only non-null values to be provided. Else ... just use `default` and save some typing :smile:

That being said if we found a lot of cases where `null` flowed into this I'd be in favor of making the argument nullable.

---
In reply to: [363547748](https://github.com/dotnet/roslyn/pull/40767#discussion_r363547748) [](ancestors = 363547748)